src/devices/machine/mm5740.cpp

```diff
@@ -138,26 +139,39 @@ TIMER_CALLBACK_MEMBER(mm5740_device::perform_scan)
 					}
 				}
 			}
-			else    // key released, unmark it from the "down" info
+			else
 			{
+				// key released, unmark it from the "down" info
 				m_x_mask[x] &= ~(1 << y);
 			}
 		}
 	}
 
+	if (m_trigger_repeat && (m_b != -1))
+	{
+		m_write_data_ready(ASSERT_LINE);
+		m_trigger_repeat = false;
+	}
+
 	if (!ako)
 	{
 		m_write_data_ready(CLEAR_LINE);
 		m_b = -1;
 	}
 }
 
+void mm5740_device::repeat_line_w(int state)
+{
+	bool repeat = (state == ASSERT_LINE);
+	m_trigger_repeat = repeat && (repeat != m_repeat);
+	m_repeat = repeat;
+}
```

Comment:
This input isn’t edge-sensitive.  The repeat input is tested on falling edges of the clock input.  You should just be storing the value here.

In `perform_scan`, which is called at the input clock frequency, you should test whether the current repeat input is high after seeing it low on the previous call to `perform_scan` and trigger a an additional data strobe (calling `m_write_data_ready`) if appropriate.